src/Symfony/Component/String/Inflector/FrenchInflector.php

```diff
@@ -0,0 +1,156 @@
+
+        // Bijou, caillou, chou, genou, hibou, joujou et pou qui prennent un x au pluriel
+        ['/^(bij|caill|ch|gen|hib|jouj|p)ou$/i', '\1oux'],
+
+        // Invariable words
+        ['/^(cinquante|soixante|mille)$/i', '\1'],
+
+        // French titles
+        ['/^(mon|ma)(sieur|dame|demoiselle|seigneur)$/', 'mes\2s'],
+        ['/^(Mon|Ma)(sieur|dame|demoiselle|seigneur)$/', 'Mes\2s'],
+    ];
+
+    /**
+     * A list of all rules for singularize.
+     */
+    private static $singularizeRegexp = [
+        // First entry: regexp
+        // Second entry: replacement
+
+        // Aspirail, bail, corail, émail, fermail, soupirail, travail, vantail et vitrail font leur pluriel en -aux
+        ['/((aspir|b|cor|ém|ferm|soupir|trav|vant|vitr))aux$/i', '\1ail'],
+
+        // Words finishing with "eau" are pluralized with a "x"
+        // Les mots finissant par "eau" prennent tous un "x" au pluriel
+        ['/(eau)x$/i', '\1'],
+
+        // Words finishing with "al" are pluralized with a "aux" expected
+        // Les mots finissant en "al" se terminent en "aux" sauf
+        ['/(amir|anim|arsen|boc|can|capit|capor|chev|crist|génér|hopit|hôpit|idé|journ|littor|loc|m|mét|minér|princip|radic|termin)aux$/i', '\1al'],
+
+        // Words finishing with "au" are pluralized with a "x" excepted "landau"
+        // Les mots finissant par "au" prennent un "x" au pluriel sauf "landau"
+        ['/(au)x$/i', '\1'],
+
+        // Words finishing with "au" are pluralized with a "x" excepted "pneu", "bleu", "émeu"
+        // Les mots finissant en "eu" prennent un "x" au pluriel sauf "pneu", "bleu", "émeu"
+        ['/(eu)x$/i', '\1'],
+
+        //  Words finishing with "ou" are pluralized with a "s" excepted bijou, caillou, chou, genou, hibou, joujou, pou
+        // Les mots finissant par "ou" prennent un "s" sauf bijou, caillou, chou, genou, hibou, joujou, pou
+        ['/(bij|caill|ch|gen|hib|jouj|p)oux$/i', '\1ou'],
+
+        // French titles
+        ['/^mes(dame|demoiselle)s$/', 'ma\1'],
+        ['/^Mes(dame|demoiselle)s$/', 'Ma\1'],
+        ['/^mes(sieur|seigneur)s$/', 'mon\1'],
+        ['/^Mes(sieur|seigneur)s$/', 'Mon\1'],
+
+        //Default rule
+        ['/s$/i', ''],
+    ];
+
+    /**
+     * A list of words which should not be inflected.
+     * This list is only used by singularize.
+     */
+    private static $uninflected = '/^(abcès|accès|abus|albatros|anchois|anglais|autobus|brebis|carquois|cas|chas|colis|concours|corps|cours|cyprès|décès|devis|discours|dos|embarras|engrais|entrelacs|excès|fois|gâchis|gars|glas|héros|intrus|jars|jus|kermès|lacis|legs|lilas|marais|mars|matelas|mépris|mets|mois|mors|obus|os|palais|paradis|parcours|pardessus|pays|plusieurs|poids|pois|pouls|printemps|processus|progrès|puits|pus|rabais|radis|recors|recours|refus|relais|remords|remous|rhinocéros|repas|rubis|sas|secours|souris|succès|talus|tapis|taudis|temps|tiers|univers|velours|verglas|vernis|virus)$/i';
```

Comment:
Some missing words: 
- bois
- fils
- rictus
- sens
- tas

Reply:
These 5 words are added. (I think that we should accept "fils". This word is certainly more use than "fil" in I.T.)